src/snowcli/cli/snowpark/commands.py

```diff
@@ -122,37 +126,60 @@ def deploy(
             "Using coverage wrapper is currently limited to project with single procedure."
         )
 
-    build_artifact_path = Path("app.zip")
+    build_artifact_path = _get_snowpark_artefact_path(snowpark)
+    zip_package = build_artifact_path.stem
+
     # TODO: this should be configurable
     if not build_artifact_path.exists():
         raise ClickException(
             "Artifact required for deploying the project does not exist in this directory. "
             "Please use build command to create it."
         )
 
+    stage_name = snowpark.get("stage_name", DEPLOYMENT_STAGE)
     stage_manager = StageManager()
     stage_manager.create(
-        stage_name=DEPLOYMENT_STAGE, comment="deployments managed by snowcli"
+        stage_name=stage_name, comment="deployments managed by snowcli"
     )
 
     packages = get_snowflake_packages()
-    deploy_status = []
+
+    artifact_stage_directory = get_app_stage_path(snowpark)
+    artifact_stage_target = f"{artifact_stage_directory}/{build_artifact_path.name}"
+
+    pm = ProcedureManager()
+
+    # Coverage case
+    if install_coverage_wrapper:
+        return _deploy_procedure_with_coverage(
+            artifact_stage_directory=artifact_stage_directory,
+            artifact_stage_target=artifact_stage_target,
+            build_artifact_path=build_artifact_path,
+            packages=packages,
+            pm=pm,
+            procedure=procedures[0],
+            replace=replace,
+            stage_manager=stage_manager,
+            stage_name=stage_name,
+        )
 
     # TODO: Check if any object already exists before we start updates
```

Comment:
Do we have any task for it? Is it still valid?

Reply:
Yes, I created a task for it